runtime/cudaq/operators.h

```diff
@@ -484,7 +492,7 @@ class sum_op {
   /// @arg `parameters` : A map of the parameter names to their concrete,
   /// complex values.
   /// @arg `invert_order`: if set to true, the ordering convention is reversed.
-  HANDLER_SPECIFIC_TEMPLATE(spin_handler)
+  PROPERTY_SPECIFIC_TEMPLATE(product_op<T>::supports_inplace_mult)
```

Comment:
Whether the handler supports inplace multiplication is a good enough proxy for whether we can/should build a sparse matrix for now. It may need to get its own property at some point, but for now I didn't see a reason to create a separate property just for that.